wp-content/themes/core/components/header/subheader_archive/Subheader_Archive_Controller.php

```diff
@@ -0,0 +1,159 @@
+		return [
+			self::CONTAINER_CLASSES => [ 'l-container' ], 
+			self::MEDIA_CLASSES     => [ 'c-subheader__media' ],
+			self::CONTENT_CLASSES   => [ 'c-subheader__content' ],
+			self::CLASSES           => [ 'c-subheader', 'c-subheader-archive' ],
+		];
+	}
+
+	public function get_classes(): string {
+		if ( has_post_thumbnail() ) {
+			$this->classes[] = 'c-subheader--has-image';
+		}
+
+		return Markup_Utils::class_attribute( $this->classes );
+	}
+
+	
+	public function get_content_classes(): string {
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	
+	public function get_media_classes(): string {
+		return Markup_Utils::class_attribute( $this->media_classes );
+	}
+
+
+	
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	public function get_attrs(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	
+	public function get_image_args(): array {
+		if (  empty( $this->hero_image_id ) ) {
+			return [];
+		}
+	
+		return [
+			Image_Controller::IMG_ID       => (int) $this->hero_image_id,
+			Image_Controller::AUTO_SHIM    => false,
+			Image_Controller::USE_LAZYLOAD => true,
+			Image_Controller::CLASSES      => [ 'c-image--overlay', 'c-image--object-fit' ],
+			Image_Controller::IMG_CLASSES  => [ 'c-subheader__media__image' ],
+			Image_Controller::SRC_SIZE     => Image_Sizes::SIXTEEN_NINE,
+			Image_Controller::SRCSET_SIZES => [
+				Image_Sizes::SIXTEEN_NINE,
+				Image_Sizes::SIXTEEN_NINE_SMALL,
+			],
+		];
+	}
+
+	
+	public function get_description_args(): array {
+		return [
+			Text_Controller::TAG     => 'p',
+			Text_Controller::CLASSES => [ 'c-subheader__description' ],
+			Text_Controller::CONTENT => $this->description,
+		];
+	}
+
+	
+	public function get_title_args(): array {
+		return [
+			Text_Controller::TAG     => 'h1',
+			Text_Controller::CLASSES => [ 'page-title', 'h1', 'c-subheader__title' ],
+			Text_Controller::CONTENT => $this->title,
+		];
+	}
+
+
+	/**
+	 * Get ACF fields 
+	 * 
+	 * Gets the content added in the meta fields 
+	 */
+	protected function map_acf_fields() {
+		if ( is_category() ) {
+			$term = get_queried_object();
+		
+			if ( ! empty( $term ) ) {
+				$this->title       = $term->name;
+				$this->description = $term->category_description;
+
+				if ( ! empty( get_field( 'hero_image', $term->taxonomy.'_'.$term->term_id ) ) ) {
+					$this->hero_image_id = get_field( 'hero_image', $term->taxonomy.'_'.$term->term_id )['ID'];
+				}
+			}
+		} else {
+			if ( ! empty( get_field( 'title', 'option' ) ) ) {
+				$this->title = get_field( 'title', 'option' );
+			}
+	
+			if ( ! empty( get_field( 'description', 'option' ) ) ) {
+				$this->description = get_field( 'description', 'option' );
+			}
+	
+			if ( ! empty( get_field( 'hero_image', 'option' ) ) ) {
+				$this->hero_image_id = get_field( 'hero_image', 'option' )['ID'];
+			}
+		}	
+	}
+
+}
```

Comment:
This should be done elsewhere, not part of this controller. We will want to be able to reuse this component for other archives (Tags, dates, etc), so the data should be passed into the component from the like other components.  Additionally, there are WP Core methods for the regular archives we should use when passing this in: `get_the_archive_title()` & `get_the_archive_description()`

Reply:
Yeah now that I think about it, this subheader is so similar the the standard subheader, we could easily combine and build this logic to handle all the scenarios. The we can let WordPress handle the various archives as expected with the option to override a taxonomy archive.

Reply:
Ok, big refact0r here. I also noticed we were handling the page title as data in the subheader as well, I moved all data management back to the route controller and then updated the subheader to be a tru component. After this I was able to remove the duplicate archive subheader.